action.yml

```diff
@@ -1,121 +1,222 @@
 branding:
   color: 'red'
   icon: 'umbrella'
 runs:
-  using: 'node20'
-  main: 'dist/index.js'
+  using: "composite"
+  steps:
+    - name: Set safe directory
+      if: ${{ inputs.set_safe_directory != 'true' }}
+      shell: bash
+      run: |
+        git config --global --add safe.directory ${{ github.workspace }}
+
+    - name: Get and set token
+      shell: bash
+      run: |
+        if [ ${{ inputs.use_oidc }} == 'true' ];
+        then
+          # {"count":1984,"value":"***"}
+          CODECOV_TOKEN=$(curl -H "Authorization: bearer $ACTIONS_ID_TOKEN_REQUEST_TOKEN" "$ACTIONS_ID_TOKEN_REQUEST_URL&audience=api://AzureADTokenExchange" | cut -d\' -f6)
+          echo "CODECOV_TOKEN=$CODECOV_TOKEN" >> $GITHUB_ENV
+        else
+          if [ -n ${{ inputs.token }} ];
+          then
+            CODECOV_TOKEN=${{ inputs.token }}
+            echo "CODECOV_TOKEN=$CODECOV_TOKEN" >> $GITHUB_ENV
+          fi
+        fi
+    - name: Upload coverage to Codecov
+      run: ./dist/codecov.sh
+      shell: bash
+      working-directory: ${{ inputs.working-directory }}
+      env:
+        CC_BINARY: ${{ inputs.binary }}
+        CC_BRANCH: ${{ inputs.override_branch }}
+        CC_BUILD: ${{ inputs.override_build }}
+        CC_BUILD_URL: ${{ inputs.override_build_url }}
+        CC_CODE: ${{ inputs.report_code }}
+        CC_DIR: ${{ inputs.directory }}
+        CC_DISABLE_FILE_FIXES: ${{ inputs.disable_file_fixes }}
+        CC_DISABLE_SEARCH: ${{ inputs.disable_search }}
+        CC_DRY_RUN: ${{ inputs.dry_run }}
+        CC_ENTERPRISE_URL: ${{ inputs.url }}
+        CC_ENV: ${{ inputs.env_vars }}
+        CC_EXCLUDES: ${{ inputs.exclude }}
+        CC_FAIL_ON_ERROR: ${{ inputs.fail_ci_if_error }}
+        CC_FILES: ${{ inputs.files }}
+        CC_FLAGS: ${{ inputs.flags }}
+        CC_GCOV_ARGS: ${{ inputs.gcov_args }}
+        CC_GCOV_EXECUTABLE: ${{ inputs.gcov_executable }}
+        CC_GCOV_IGNORE: ${{ inputs.gcov_ignore }}
+        CC_GCOV_INCLUDE: ${{ inputs.gcov_include }}
+        CC_GIT_SERVICE: ${{ inputs.git_service }}
+        CC_HANDLE_NO_REPORTS_FOUND: ${{ inputs.handle_no_reports_found }}
+        CC_JOB_CODE: ${{ inputs.job_code }}
+        CC_LEGACY: ${{ inputs.use_legacy_upload_endpoint }}
+        CC_NAME: ${{ inputs.name }}
+        CC_NETWORK_FILTER: ${{ inputs.network_filter }}
+        CC_NETWORK_PREFIX: ${{ inputs.network_prefix }}
+        CC_NETWORK_ROOT_FOLDER: ${{ inputs.root_dir }}
+        CC_OS: ${{ inputs.os }}
+        CC_PARENT_SHA: ${{ inputs.commit_parent }}
+        CC_PLUGINS: ${{ inputs.plugins }}
+        CC_PR: ${{ inputs.override_pr }}
+        CC_REPORT_TYPE: ${{ inputs.report_type }}
+        CC_SHA: ${{ inputs.override_commit }}
+        CC_SKIP_VALIDATION: ${{ inputs.skip_validation }}
+        CC_SLUG: ${{ inputs.slug }}
+        CC_SWIFT_PROJECT: ${{ inputs.swift_project }}
+        CC_TOKEN: $CODECOV_TOKEN
+        CC_VERBOSE: ${{ inputs.verbose }}
+        CC_VERSION: ${{ inputs.version }}
```

Comment:
I don't know if these are new, but may want to confirm there is no overlap with `cc` envs. (If such exist)

Reply:
I think these should be safe!